src/main/java/htsjdk/samtools/util/OverlapDetector.java

```diff
@@ -42,6 +42,8 @@
     private final int lhsBuffer;
     private final int rhsBuffer;
 
+    private Set<T> allObjectsCache = null;
```

Comment:
`= null` is unnecessary here since that's the value the field already has

Reply:
I'm inclined to leave the assignment for clarity, unless you really object.

Reply:
If we want to include this I think it would be a good idea to make this be 
- 1 a `transient`field (even thought it's not serializable, I think it's a good idea to mark it so it's clear)
- 2 possibly make it into a `SoftReference` so that it won't blow up the memory requirements.  The overlap detector can potentially be storing a rather large set of things, and unintentionally doubling it could be problematic.